auth_options.go

```diff
@@ -194,8 +209,35 @@ func (opts *AuthOptions) ToTokenV3CreateMap(scope map[string]interface{}) (map[s
 			req.Auth.Identity.Token = &tokenReq{
 				ID: opts.TokenID,
 			}
+
+		} else if opts.ApplicationCredentialID != "" {
+			// Configure the request for ApplicationCredentialID authentication.
+			// https://github.com/openstack/keystone-specs/blob/master/specs/keystone/queens/application-credentials.rst#using-an-application-credential-to-obtain-a-token
+			// There are three kinds of possible application_credential requests
+			// 1. application_credential id + secret.
+			// 2. application_credential name + user_id + secret
+			// 3. application_credential name + user_name + user_domain_id/user_domain_name
+			req.Auth.Identity.Methods = []string{"application_credential"}
+			req.Auth.Identity.ApplicationCredential = &applicationCredentialReq{
+				ID:     &opts.ApplicationCredentialID,
+				Secret: &opts.ApplicationCredentialSecret,
+			}
+		} else if opts.ApplicationCredentialName != "" {
+			req.Auth.Identity.Methods = []string{"application_credential"}
+			if opts.DomainID != "" {
+				userRequest = userReq{Name: &opts.Username, Domain: &domainReq{ID: &opts.DomainID}}
+			} else if opts.DomainName != "" {
+				userRequest = userReq{Name: &opts.Username, Domain: &domainReq{Name: &opts.DomainName}}
+			} else {
+				userRequest = userReq{Name: &opts.Username}
```

Comment:
I don't think this variation is possible?

Reply:
Probably you are right. @RdL87 I will drop this

Reply:
Yes you are right this cannot work without domain. If we had implemented using UsedID that it would have make sense to create a request without DomainID/Name, but it is not our case.